pkg/metrics/metrics.go

```diff
@@ -256,25 +216,131 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
+
+	// remove stale service objects
+	_, err = c.CoreV1().Services(c.namespace).Get(ctx, staleResourceName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "serviceRetrievalFailed",
+			"Stale service retrieval failure")
+		log.Error(err, "error getting stale service object")
+	}
+	if err == nil {
+		if err := c.CoreV1().Services(c.namespace).Delete(ctx, staleResourceName, metav1.DeleteOptions{}); err != nil {
+			c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "serviceDeletionFailed",
+				"Stale service deletion failure")
+			log.Error(err, "error deleting stale service object")
+		}
+	}
+
+	// remove stale endpoint objects
+	_, err = c.CoreV1().Endpoints(c.namespace).Get(ctx, staleResourceName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "endpointRetrievalFailed",
+			"Stale endpoint retrieval failure")
+		log.Error(err, "error getting stale endpoint object")
+	}
+	if err == nil {
+		if err := c.CoreV1().Endpoints(c.namespace).Delete(ctx, staleResourceName, metav1.DeleteOptions{}); err != nil {
+			c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "endpointDeletionFailed",
+				"Stale endpoint deletion failure")
+			log.Error(err, "error deleting stale endpoint object")
+		}
+	}
+
+	// remove stale serviceMonitor objects
+	_, err = c.ServiceMonitors(c.namespace).Get(ctx, staleResourceName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "serviceMonitorRetrievalFailed",
+			"Stale serviceMonitor retrieval failure")
+		log.Error(err, "error getting stale serviceMonitor object")
+	}
+	if err == nil {
+		if err := c.ServiceMonitors(c.namespace).Delete(ctx, staleResourceName, metav1.DeleteOptions{}); err != nil {
+			c.recorder.Event(wmcoNamespace, core.EventTypeWarning, "serviceMonitorDeletionFailed",
+				"Stale serviceMonitor deletion failure")
+			log.Error(err, "error deleting stale serviceMonitor object")
+		}
+	}
+}
+
+// createEndpoint creates an endpoint object in the operator namespace.
+// WMCO is no longer creating a service with a selector therefore no Endpoint
+// object is created and WMCO needs to create the Endpoint object.
+// We cannot create endpoints as a part of manifests deployment as
+// Endpoints resources are not currently OLM-supported for bundle creation.
+func (c *Config) createEndpoint() error {
+	// create new Endpoint
+	newEndpoint := &v1.Endpoints{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Endpoints",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      WindowsMetricsResource,
+			Namespace: c.namespace,
+			Labels:    map[string]string{"name": WindowsMetricsResource},
+		},
+		Subsets: nil,
+	}
+	_, err := c.CoreV1().Endpoints(c.namespace).Get(context.TODO(),
+		WindowsMetricsResource, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
Don't we need to delete and recreate the endpoint to ensure we are not using a stale `windows-exporter` resource created by a previous version 2.0 version of the operator?